### PR TITLE
AP-746 Generate means report PDF

### PR DESCRIPTION
**Pull request by Kalagan**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-746)

Generate a means report at the end of the application.
The PDF is generated in the background, saved to S3 via ActiveStorage and attached to the application.
The PDF is based on an HTML page similar to a check answers page.
You can access it via `/providers/applications/:uuid/means_report`

## Checklist

Before you ask people to review this PR:

- [X] Tests and rubocop should be passing: `bundle exec rake`
- [X] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [X] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [X] The PR description should say what you changed and why, with a link to the JIRA story.
- [X] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [X] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/shared/_client_details_v2.html.erb [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (mpw5) [inline app/views/shared/_client_details_v2.html.erb]: Do we need the `_v2` in the name of this?

File: spec/requests/providers/merits_declarations_spec.rb [hunk 2]
```diff
@@ -41,19 +41,19 @@
...
 
         it 'redirects to next page' do
           subject
           expect(response).to redirect_to(providers_legal_aid_application_end_of_application_path(legal_aid_application))
         end
 
-        it 'creates a merits report pdf' do
-          MeritsReportCreatorWorker.clear
-          expect(MeritsReportCreator).to receive(:call).with(legal_aid_application)
+        it 'creates a pdf reports' do
```
Comment (mpw5) [inline spec/requests/providers/merits_declarations_spec.rb]: Being really picky, this should probably be `creates pdf reports` or `creates a pdf report`.

File: spec/services/reports/reports_creator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Reports::ReportsCreator do
+  let(:legal_aid_application) { create :legal_aid_application, :with_proceeding_types, :with_everything, state: :generating_reports }
+
+  subject { described_class.call(legal_aid_application) }
+
+  describe '.call' do
+    it 'attaches merits_report.pdf to the application' do
```
Comment (mpw5) [inline spec/services/reports/reports_creator_spec.rb]: As this tests for both reports, would `attaches merits_report.pdf and means_report.pdf to the application` be more accurate?

Comment (Kalagan) [reply spec/services/reports/reports_creator_spec.rb]: yes definitely 👍

File: app/views/shared/_client_details_v2.html.erb (hunk 1 above)
Comment (Kalagan) [reply app/views/shared/_client_details_v2.html.erb]: it's because there's already a `_client_details.html.erb` but it shows different information :/

Comment (mpw5) [reply app/views/shared/_client_details_v2.html.erb]: 👌

**Pull request merged**